src/main/java/core/basesyntax/strategy/FruitTransactionStrategy.java

```diff
@@ -0,0 +1,15 @@
+package core.basesyntax.strategy;
+
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.service.FruitOperationService;
+import core.basesyntax.service.impl.FruitPurchaseService;
+import core.basesyntax.service.impl.FruitSupplyService;
+
+public class FruitTransactionStrategy {
+    public FruitOperationService getFruitOperationService(FruitTransaction.Operation operation) {
+        if (operation == FruitTransaction.Operation.PURCHASE) {
+            return new FruitPurchaseService();
+        }
+        return new FruitSupplyService();
```

Comment:
let's make a separate handler for each operation. It may look like overhead but it's a more flexible solution. if tomorrow will be a feature request to process BALANCE transactions differently - changes will only affect only one balance handler.

Let's make strategy to contain mapping of operation to handler and return this handler